tools/setupResourceScripts/setup_tdr_resources.py

```diff
@@ -1,213 +1,278 @@
+    print(f"Deleted dataset {filtered_datasets[0].id}")
+
+
+def add_snapshot_builder_settings(clients, id, directory, snapshot_builder_settings_file):
+  with open(
+    os.path.join("files", directory, snapshot_builder_settings_file)) as dataset_schema_json:
+    # In order to use the client API, we need to publish the updated python client library
+    dataset = requests.post(f"{clients.api_client.configuration.host}/api/repository/v1/datasets/{id}/snapshotBuilder/settings", json=json.load(dataset_schema_json), auth=(
+      'bearer', clients.api_client.configuration.access_token))
+  return dataset
 
 
 def main():
-    parser = argparse.ArgumentParser()
-    parser.add_argument('--host', default='https://jade-4.datarepo-integration.broadinstitute.org')
-    parser.add_argument('--datasets', default='./suites/datarepo_datasets.json')
-    parser.add_argument('--gcp_profile_id')
-    parser.add_argument('--azure_profile_id')
-    args = parser.parse_args()
-    clients = Clients(args.host)
-
-    add_jade_stewards = 'dev' in args.host or 'integration' in args.host
-    gcp_profile_id = args.gcp_profile_id
-    azure_profile_id = args.azure_profile_id
-
-    outputs = []
-    for dataset_to_upload in get_datasets_to_upload(args.datasets):
-        dataset_cloud_platform = dataset_to_upload['cloud_platform']
-        if dataset_cloud_platform == 'gcp':
-            profile_id = gcp_profile_id
-        elif dataset_cloud_platform == 'azure':
-            profile_id = azure_profile_id
-        else:
-            raise Exception("Billing profile create not yet supported for cloud platform: " + dataset_cloud_platform)
-        if profile_id is None:
-            profile_job_response = create_billing_profile(clients, add_jade_stewards, dataset_cloud_platform)
-            profile_id = profile_job_response['id']
-        created_dataset = create_dataset(clients, dataset_to_upload, profile_id)
-        dataset_name = created_dataset['name']
-        output_ids = {dataset_name: {'dataset_id': created_dataset['id']}}
-        if dataset_to_upload.get('snapshots'):
-            snapshot_ids = create_snapshots(clients, dataset_to_upload['name'], dataset_to_upload['snapshots'],
-                                            profile_id)
-            output_ids[dataset_name]['snapshot_ids'] = snapshot_ids
-            outputs.append(output_ids)
-
-    output_filename = f"{os.path.basename(args.datasets).split('.')[0]}_outputs.json"
-    with open(output_filename, 'w') as f:
-        json.dump(outputs, f)
-
-
-if __name__ == "__main__":
-    main()
+  parser = argparse.ArgumentParser()
+  parser.add_argument('--host', default='https://jade-4.datarepo-integration.broadinstitute.org')
+  parser.add_argument('--datasets', default='./suites/datarepo_datasets.json')
+  parser.add_argument('--gcp_profile_id')
+  parser.add_argument('--azure_profile_id')
+  parser.add_argument('--azure_managed_app_name')
+  parser.add_argument('--billing_profile_file_name')
```

Comment:
What do you think about updating the README with these new arguments? https://github.com/DataBiosphere/jade-data-repo/blob/develop/tools/setupResourceScripts/README.md